cpp/src/robotick/framework/data/TelemetryServer.cpp

```
@@ -0,0 +1,348 @@
+		nlohmann::ordered_json struct_json;
+		struct_json["type"] = struct_desc ? struct_desc->name.c_str() : "null";
+		struct_json["offset_within_container"] = static_cast<int>(base_offset);
+
+		workload_json[struct_name] = struct_json;
+
+		void* data_ptr = (uint8_t*)workload_ptr + base_offset;
+
+		emit_type_info(layout_json, workloads_buffer, data_ptr, struct_desc);
+	}
+
+	static size_t get_process_memory_used()
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)